theseus/optimizer/autograd/lu_cuda_sparse_autograd.py

```diff
@@ -5,15 +5,26 @@
 
 import torch
 
-from theseus.extlib.cusolver_lu_solver import CusolverLUSolver
-from theseus.extlib.mat_mult import apply_damping, mat_vec, mult_MtM, tmat_vec
-
 from ..linear_system import SparseStructure
 
 
 class LUCudaSolveFunction(torch.autograd.Function):
     @staticmethod
     def forward(ctx, *args, **kwargs):
+        if not torch.cuda.is_available():
```

Comment:
Any reason we cannot move all of these conditional imports to the top of the file?

Reply:
Well if so we are back to having to implement a mock-class for when cuda is not available...

Reply:
I'm mostly wondering about any overheads from repeated `import` calls. I know python caches the module for future imports, but there might some additional non-negligible cost for calling `import`? (not sure). We could get around writing a mock class by doing the conditional import at the top while also setting a boolean flag, say `_EXTLIB_WAS_LOADED`, and (inside the class) throw the runtime error based on the status of the flag (rather than  checking the import).

Reply:
I see - yeah that would work. I would go for the cleaner solution, just not sure which is (ideas @mhmukadam ?) if there is an overhead doing import it will be certainly very small, and I feel we should not optimize around something that should already be optimized inside python.

Reply:
I checked with other engineers, and importing does have some overhead, but perhaps it's negligible. Since we are doing benchmarking this half anyway, I vote for keeping this version, and revisit the issue if we see opportunities for improvement later. Sounds good?

Reply:
Sounds good. I would have proposed the same thing -- keep this and during benchmarking see if this is a bottle in performance to be prioritize and fixed.